src/components/Name.js

```diff
@@ -0,0 +1,37 @@
+import React from 'react';
+import {connect} from 'react-redux';
+import Input from './Input';
+import * as selectors from '../redux/selectors';
+
+export class Save extends React.Component {
+  render() {
+    if (!this.props.hasConnectedDrive) {
+      return null;
+    }
+    if (this.props.isSavingDrive) {
+      return (
+        <Input placeholder="Program Name"/>
+      );
+    }
+    return (
+      <Input placeholder="Program Name"/>
+    );
+  }
+}
+
+
+
+Save.propTypes = {
+  hasConnectedDrive: React.PropTypes.bool,
+  isSavingDrive: React.PropTypes.bool,
+  hasSavedDrive: React.PropTypes.bool,
+  saveGoogleDrive: React.PropTypes.func,
```

Comment:
these last two props look like dead code. I would say either leave a `// TODO` to use them, or delete them until you are ready to implement.